test/new-e2e/tests/agent-shared-components/api/api_test.go

```diff
@@ -70,32 +73,237 @@ func (endpointInfo *agentEndpointInfo) fetchCommand(authtoken string) string {
 	)
 }
 
-func (v *apiSuite) TestInternalAgentAPIEndpoints() {
+func (v *apiSuite) TestDefaultAgentAPIEndpoints() {
 	testcases := []struct {
 		agentEndpointInfo
+		name string
 		want string
 	}{
 		{
+			name: "version",
 			agentEndpointInfo: agentEndpointInfo{
-				name:     "version",
 				scheme:   "https",
 				port:     agentCmdPort,
 				endpoint: "/agent/version",
 				method:   "GET",
 				data:     "",
 			},
+			// TODO: json parse this to a better comparison
 			want: `"Major":7,"Minor":5`,
 		},
 		{
+			name: "hostname",
+			agentEndpointInfo: agentEndpointInfo{
+				scheme:   "https",
+				port:     agentCmdPort,
+				endpoint: "/agent/hostname",
+				method:   "GET",
+				data:     "",
+			},
+			// ec2 instance id's start with i-
+			want: `i-`,
+		},
+		{
+			name: "health",
+			agentEndpointInfo: agentEndpointInfo{
+				scheme:   "https",
+				port:     agentCmdPort,
+				endpoint: "/agent/status/health",
+				method:   "GET",
+				data:     "",
+			},
+			// TODO: json parse this to a better comparison
+			want: `{"Healthy":[`,
+		},
+		// TODO: we could possibly do a regexp comparison
+		// {
+		// 	name: "csrf-token",
+		// 	agentEndpointInfo: agentEndpointInfo{
+		// 		scheme:   "https",
+		// 		port:     agentCmdPort,
+		// 		endpoint: "/agent/gui/csrf-token",
+		// 		method:   "GET",
+		// 		data:     "",
+		// 	},
+		// 	want: ``,
+		// },
+		{
+			name: "config",
+			agentEndpointInfo: agentEndpointInfo{
+				scheme:   "https",
+				port:     agentCmdPort,
+				endpoint: "/agent/config",
+				method:   "GET",
+				data:     "",
+			},
+			// TODO: find a better setting to compare output to
+			want: `api_key: '*******`,
```

Comment:
👍 I was thinking about maybe doing this and having a func with the assertion as part of the testcase, something like:
```
func(t *testing, resp string) { assert.Contains(t, resp, api_key) }
```

this would help solve the issue where some responses are strings and some are json so we can do https://github.com/DataDog/datadog-agent/pull/25000#discussion_r1576414244 too

Reply:
fixed in https://github.com/DataDog/datadog-agent/pull/25000/commits/31cbfb387bc2003cc5cbd75d97162e2e2228a727